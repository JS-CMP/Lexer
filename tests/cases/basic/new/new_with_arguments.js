function Greeter(name) {
    this.greet = function () {
        return "Hello, " + name;
    };
}

new Greeter("World").greet(); // NEW(Greeter, "World").greet();
