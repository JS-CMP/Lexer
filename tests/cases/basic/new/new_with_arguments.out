#include "types/JsAny.hpp"
#include "global/global.hpp"
#include "types/objects/Types.hpp"
#include "customOperators/CustomOperators.hpp"

int main() {JS::Any Greeter;Greeter = JS::Any(std::make_shared<JS::Function>([&](const JS::Any &thisArg, const JS::Any &arguments) -> JS::Any {JS::Any name = arguments["0"];thisArg[u"greet"]=JS::Any(std::make_shared<JS::Function>([&](const JS::Any &thisArg, const JS::Any &arguments) -> JS::Any {return JS::Any(u"Hello");return JS::Any();}));return JS::Any();}));console[u"log"](NEW(Greeter,JS::Any(u"World"))[u"greet"]());}
